pkg/runtime/wfengine/activity.go

```diff
@@ -146,7 +146,7 @@ func (a *activityActor) executeActivity(ctx context.Context, actorID string, nam
 		return err
 	}
 
-	endIndex := strings.LastIndex(actorID, "#")
+	endIndex := strings.LastIndex(actorID, "_")
```

Comment:
I am not very much aware if || can cause issue in some database, just like # did. Till now, I believe, was following the comment https://github.com/dapr/dapr/issues/6156#issuecomment-1492240506.
cc @cgillum Thoughts here?

Reply:
@artursouza I think using `||` might be too much risk in terms of compatibility for minimal benefit.

Currently the `||` delimiter is used by Dapr's actor subsystem for keys in the form `{appName}||{actorType}||{actorID}||{key}`. My concern is that if we embed another `||` delimiter into `{actorID}`, then it could cause other parts of the system (or components) to incorrectly parse actor state keys if they make assumptions about the positions of segments between the `||` delimeters. To mitigate the potential damage of those kinds of assumptions in other code, I'd prefer to play it safe and use a separate delimiter that is specific to workflow actor IDs.

Reply:
We need a better delimiter. _ is too commonly used by apps.

Reply:
How about `::`?  That should be acceptable to Cosmos DB.

Reply:
I have updated PR with `::` for now.